Radius & Pi

In [1]:
r <- 2
pi <- 3.14159

A function to compute the volume of a sphere with radius r=2

In [2]:
volume <- function(r) {
  4/3*pi*r^3
}

volume(r)

[1] 33.51029

Now lets see if our function is performing properly by changing value of r (the result should be approx. 904.78):

In [3]:
4/3*3.14159*6^3

[1] 904.7779

A function to compute the volumes of the spheres with radius r, r^2 and r^3


In [4]:
volume_vector <- function(r){
  for (i in 1:3) {
    v <- 4/3*pi
    r <- 22^3
    a <- c(1,2,3)
    result <- v*r^a
    return(result)
  }
}

Run volume_vector(r) and print the volumes of the spheres with radius r, r^2 and r^3

In [5]:
volume_vector(r)

[1] 4.460220e+04 4.749242e+08 5.056993e+12

Again, lets test the function (result should be approx. 44.406, 474.925, and 5.057)

In [6]:
4/3*3.14159*22^3^1
4/3*3.14159*(22^3)^2
4/3*3.14159*(22^3)^3

[1] 44602.2

[1] 474924230

[1] 5.056993e+12

We have now fixed the bugs in the code, and it should run as intended.
